frontend/src/modules/application/components/ApplicationForm.jsx

```diff
@@ -14,64 +22,65 @@ const TextboxInput = ({ label, curValue, onChange, required }) => <div>
               required={required}/>
 </div>
 
-const BinaryRadioInput = ({ label, curValue,required }) => <div>
+// Binary radio input field component
+const BinaryRadioInput = ({ label, curValue, onChange, required }) => <div>
     <label style={{ width: "20vw" }}>{required ? label + ':' : label}</label>
     <label>
-        <input type="radio" checked={curValue === "true"}/>
+        <input type="radio" value="true" checked={curValue === "true"} onChange={onChange}/>
         Yes
     </label>
     <label>
-        <input type="radio" checked={curValue === "false"}/>
+        <input type="radio" value="false" checked={curValue === "false"} onChange={onChange}/>
         No
     </label>
 </div>
 
-
-
-const allFields = [].concat.apply([], 
-    [personalInformationFields, 
-    currentProgramInformationFields,
-    currentStatusFields, 
-    customQuestions])
-
-const initialState = allFields.reduce((acc, cur) => ({ ...acc, [cur.value]: "" }), {})
-
+// Application form component that holds multiple individual simple components defined earlier
 class ApplicationForm extends React.Component {
 
-    state = { ...initialState }
-
-    handleChange = prop => event => this.setState({ [prop]: event.target.value })
+    // dispatches a redux action to update the store
+    handleChange = field_name => event => {
+        this.props.updateField({ "key": field_name,
+                                "value": event.target.value})
+    }
 
+    // TODO: used for submitting the form and saving it in the backend
```

Comment:
Im ok with leaving a todo in the comments, but can we make sure that an associated issue gets created for it ? That's easier for the team to track imo.